actors/user/src/main/java/org/sunbird/user/actors/UserRoleActor.java

```diff
@@ -0,0 +1,355 @@
+      for (Map<String, Object> map : list) {
+        Map<String, Object> roleResponseMap = new HashMap<>();
+        roleResponseMap.put(JsonKey.ID, map.get(JsonKey.ID));
+        roleResponseMap.put(JsonKey.NAME, map.get(JsonKey.NAME));
+        List<String> roleGroup = (List) map.get(JsonKey.ROLE_GROUP_ID);
+        List<Map<String, Object>> actionGroupListMap = new ArrayList<>();
+        roleResponseMap.put(JsonKey.ACTION_GROUPS, actionGroupListMap);
+        Map<String, Object> subRoleResponseMap = null;
+        for (String val : roleGroup) {
+          subRoleResponseMap = new HashMap<>();
+          Map<String, Object> subRoleMap = getSubRoleListMap(roleGroupMap, val);
+          List<String> subRole = (List) subRoleMap.get(JsonKey.URL_ACTION_ID);
+          List<Map<String, Object>> roleUrlResponList = new ArrayList<>();
+          subRoleResponseMap.put(JsonKey.ID, subRoleMap.get(JsonKey.ID));
+          subRoleResponseMap.put(JsonKey.NAME, subRoleMap.get(JsonKey.NAME));
+          for (String rolemap : subRole) {
+            roleUrlResponList.add(getRoleAction(urlActionListMap, rolemap));
+          }
+          if (subRoleResponseMap.containsKey(JsonKey.ACTIONS)) {
+            List<Map<String, Object>> listOfMap =
+                (List<Map<String, Object>>) subRoleResponseMap.get(JsonKey.ACTIONS);
+            listOfMap.addAll(roleUrlResponList);
+          } else {
+            subRoleResponseMap.put(JsonKey.ACTIONS, roleUrlResponList);
+          }
+          actionGroupListMap.add(subRoleResponseMap);
+        }
+
+        resposnemap.add(roleResponseMap);
+      }
+    }
+    mergeResponse.getResult().put(JsonKey.ROLES, resposnemap);
+    sender().tell(mergeResponse, self());
+  }
+
+  /**
+   * This method will assign roles to users or user organizations.
+   *
+   * @param actorMessage
+   */
+  @SuppressWarnings("unchecked")
+  private void assignRoles(Request actorMessage) {
+    Map<String, Object> requestMap = actorMessage.getRequest();
+
+    if (null != requestMap.get(JsonKey.ROLES)
+        && !((List<String>) requestMap.get(JsonKey.ROLES)).isEmpty()) {
+      String msg = Util.validateRoles((List<String>) requestMap.get(JsonKey.ROLES));
+      if (!msg.equalsIgnoreCase(JsonKey.SUCCESS)) {
+        throw new ProjectCommonException(
+            ResponseCode.invalidRole.getErrorCode(),
+            ResponseCode.invalidRole.getErrorMessage(),
+            ResponseCode.CLIENT_ERROR.getResponseCode());
+      }
+    }
+    // object of telemetry event...
+    String userId = (String) requestMap.get(JsonKey.USER_ID);
+    String externalId = (String) requestMap.get(JsonKey.EXTERNAL_ID);
+    String provider = (String) requestMap.get(JsonKey.PROVIDER);
+    String organisationId = (String) requestMap.get(JsonKey.ORGANISATION_ID);
+    String hashTagId = null;
+    Map<String, Object> map = null;
+    // have a check if organisation id is provided then need to get hashtagId
+    if (StringUtils.isNotBlank(organisationId)) {
```

Comment:
User OrganisationClient (being developed by Rajat) to get organisation details for given org ID. Ignore external ID and provider for now to fetch org details.